plugins/modules/ome_alert_policies.py

```diff
@@ -0,0 +1,940 @@
+            if start_time_x < datetime.now():
+                module.exit_json(
+                    failed=True, msg="Start time must be greater than current time.")
+            schedule_payload["StartTime"] = start_time
+        except ValueError:
+            module.exit_json(failed=True, msg="Invalid start date or time.")
+        if inp_schedule.get('date_to'):
+            end_time = f"{inp_schedule.get('date_to')} {time_to}:00.000"
+            try:
+                end_time_x = datetime.strptime(end_time, time_format)
+                if end_time_x < start_time_x:
+                    module.exit_json(
+                        failed=True, msg="End time must be greater than start time.")
+                schedule_payload["EndTime"] = end_time
+            except ValueError:
+                module.exit_json(failed=True, msg="Invalid end date or time.")
+        weekdays = {'monday': 'mon', 'tuesday': 'tue', 'wednesday': 'wed', 'thursday': 'thu', 'friday': 'fri',
+                    'saturday': 'sat', 'sunday': 'sun'}
+        inp_week_list = ['*']
+        if inp_schedule.get('days'):
+            inp_week_list = set(inp_schedule.get('days'))
+        schedule_payload["CronString"] = f"* * * ? * {SEPARATOR.join([weekdays.get(x, '*') for x in inp_week_list])} *"
+    return {"Schedule": schedule_payload}
+
+
+def get_actions_payload(module, rest_obj):
+    """
+    Generates the payload for the actions to be performed.
+
+    Args:
+        module (object): The module object.
+        rest_obj (object): The REST object.
+
+    Returns:
+        dict: The dictionary containing the actions payload.
+    """
+    action_payload = []
+    inp_actions = module.params.get('actions')
+    if inp_actions:
+        ref_actions = get_all_actions(rest_obj)
+        inp_dict = dict((x.get("action_name"), dict((y.get("name"), y.get(
+            "value")) for y in x.get("parameters", []))) for x in inp_actions)
+        if 'Ignore' in inp_dict:
+            pld = {}
+            pld['TemplateId'] = ref_actions.get('Ignore').get('Id')
+            pld['Name'] = "Ignore"
+            pld['ParameterDetails'] = []
+            action_payload.append(pld)
+        else:
+            for inp_k, inp_val in inp_dict.items():
+                if inp_k in ref_actions:
+                    pld = {}
+                    if ref_actions.get(inp_k).get('Disabled'):
+                        module.exit_json(
+                            failed=True, msg=f"Action '{inp_k}' is disabled.")
+                    pld['TemplateId'] = ref_actions.get(inp_k).get('Id')
+                    pld['Name'] = inp_k
+                    diff = set(inp_val.keys()) - \
+                        set(ref_actions.get(inp_k).get('Parameters').keys())
+                    if diff:
+                        module.exit_json(
+                            failed=True, msg=f"Action '{inp_k}' has invalid parameters: {SEPARATOR.join(diff)}")
```

Comment:
Avoid single quote in between the sentence, it's not easy to assert same during automation.

Reply:
same comments to other places.

Reply:
updated